akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala

```diff
@@ -193,12 +169,9 @@ private[akka] class TLSActor(settings: ActorMaterializerSettings,
       case Some(TLSClientAuth.Need) ⇒ engine.setNeedClientAuth(true)
       case _                        ⇒ // do nothing
     }
-    params.sslParameters foreach { p ⇒
-      //first copy the mutable SLLParameters before modifying to prevent race condition
-      val parameters = cloneSSLParameters(p)
-      hostInfo foreach { case (host, _) ⇒ applySNI(host, parameters) }
-      engine.setSSLParameters(parameters)
-    }
+
+    // configure Server Name Indication unless ssl-config disabled it (in which case we already logged many warnings)
+    applySNI(params)
```

Comment:
Yeah, we discussed this at the point where this was introduced, it's a bit tricky since "where to call it" is half of the problem here.

Previously we could not have put it directly into ssl-config (it would have to use reflection or other hackery) since it was JDK6 supporting, yesterday I moved ssl-config 0.2 to JDK8 in which the `setServerNames` method is available so we could do that move... I'll open a ticket on ssl-config